tests/e2e/egress/egress_test.go

```diff
@@ -57,400 +48,132 @@ func TestE2EEgressConnectivity(t *testing.T) {
+			cmd:                []string{"curl", "-sSL", "-m", "10", "--fail-with-body", "https://httpbin.org/headers"},
+			applyEgressConfig:  true,
+			applyNetworkPolicy: true,
+			expectError:        false,
 		},
 		{
-			name:                 "connection to kyma-project is refused when NetworkPolicy is applied and egress is configured",
-			cmd:                  []string{"curl", "-sSL", "-m", "10", "https://kyma-project.io"},
-			enableIstioInjection: true,
-			applyEgressConfig:    true,
-			applyNetworkPolicy:   true,
-			expectError:          true,
+			name:               "connection to kyma-project is refused when NetworkPolicy is applied and egress is configured",
+			cmd:                []string{"curl", "-sSL", "-m", "10", "--fail-with-body", "https://kyma-project.io"},
+			applyEgressConfig:  true,
+			applyNetworkPolicy: true,
+			expectError:        true,
 		},
 	}
 
 	for _, tt := range tc {
 		t.Run(tt.name, func(t *testing.T) {
-			ns := createTestNamespace(ctx, c, tt.enableIstioInjection, t)
+			runID := envconf.RandomName("e2e-egress", 16)
+			// instantiate a resources client
+			r, tErr := resources.New(cfg.Client().RESTConfig())
+			require.NoError(t, tErr)
+
+			// namespace creation
+			require.NoError(t, ns.CreateNamespace(ctx, t, runID, cfg,
+				ns.WithLabels(map[string]string{"istio-injection": "enabled"}),
+			))
 			if tt.applyEgressConfig {
-				err := applyEgressConfig(ctx, c, ns)
-				if err != nil {
-					t.Errorf("failed to create egress resources: %v", err)
-				}
+				t.Log("Applying egress config for the test: ", runID)
+				require.NoError(t, decoder.DecodeEachFile(ctx, fsys, "egress_config.yaml", decoder.CreateHandler(r), decoder.MutateNamespace(runID)))
 			}
 			if tt.applyNetworkPolicy {
-				applyNetworPolicy(ctx, c, t, ns)
+				t.Log("Applying network policy for the test: ", runID)
+				require.NoError(t, decoder.DecodeEachFile(ctx, fsys, "networkpolicy.yaml", decoder.CreateHandler(r), decoder.MutateNamespace(runID)))
 			}
 
-			out, err := runCurlInPod(ctx, cfg, ns, tt.cmd)
+			// test using pod with curl
+			testPod := &corev1.Pod{}
+			require.NoError(t, decoder.DecodeFile(fsys, "curl_pod.yaml", testPod, decoder.MutateNamespace(runID)))
+			require.NoError(t, r.Create(ctx, testPod))
+			require.NoError(t, wait.For(conditions.New(r).PodRunning(testPod)))
+
+			var stdout, stderr bytes.Buffer
+			err := r.ExecInPod(ctx, runID, testPod.Name, "curl", tt.cmd, &stdout, &stderr)
 			if err != nil && !tt.expectError {
 				t.Errorf("got an error but shouldn't have: %v", err)
 			}
 			if err == nil && tt.expectError {
 				t.Error("didn't get an error but expected one")
 			}
 
-			t.Log(string(out))
+			t.Log(stdout.String())
+			t.Log(stderr.String())
 		})
 	}
 }
 
-func setupIstio(ctx context.Context, c client.Client, t *testing.T) {
-	// this function is a helper
+func setupIstio(ctx context.Context, fsys fs.FS, cfg *envconf.Config, t *testing.T) error {
```

Comment:
This function will not be possible to be reused in the current form.
This is caused by:
- It uses a local `istio_customresource.yaml` instead of the release artifact `https://github.com/kyma-project/istio/releases/latest/download/istio_customresource.yaml`
- It does not create the istio manager resources

Additionally it has a dependency to structured `operatorv1alpha2.Istio` API, which will become a unnecessary dependency if we were to use it in API Gateway

Reply:
Another thing is that it probably should use `make deploy` (or something equivalent) in case of installing the module from the same repo (so installing Istio here, and APIGateway if such a helper would exist for APIGateway) as we have the need to test on the current image.